src/renderer/ContentTypes.ts

```
@@ -36,6 +36,16 @@ const defaultRegistry: {
   [K in Context]?: Component
 } = {}
 
+let repo: RepoFrontend
+
+export function terribleRepoBorrowingForWorkspace() {
+  return repo
+}
+
+export function initRepo(repoIn: RepoFrontend) {
```

Maybe `__setRepo` and `__getRepo` with appropriate warning comments and use `__getRepo` in the functions below (rather than accessing `repo` directly). `__getRepo` should throw an error if repo isn't set.